### ISSUE TITLE
selecthighlight doesn't match when wrap option is on

### ISSUE THREAD

**Issue by OsicKwon** [lite-xl/lite-xl-plugins#396]
It works very well without wrap.
![image](https://github.com/lite-xl/lite-xl-plugins/assets/5477324/ad74cc2c-0c71-4efd-ad91-12ea9b9d4eb7)

However, when wrap is enabled, it looks weired like below.
![image](https://github.com/lite-xl/lite-xl-plugins/assets/5477324/89e7c6cf-66a2-45f8-9b57-cd7566d529be)

**Comment by Guldoman** [comment]
Yeah, at the moment the way `linewrapping` is implemented is a bit problematic. We're working on rewriting that system.
